pallets/elections/src/lib.rs

```diff
@@ -443,26 +459,46 @@ pub mod pallet {
         type Error = ElectionError;
         type DataProvider = T::DataProvider;
 
-        /// The elections are PoA so only the nodes listed in the Validators will be elected as
-        /// validators.
-        ///
-        /// We calculate the supports for them for the sake of eras payouts.
+        /// We calculate the supports for each validator. The external validators are chosen as:
+        /// 1) "`NextEraNonReservedValidators` that are staking and are not banned" in case of Permissioned ElectionOpenness
+        /// 2) "All staking and not banned validators" in case of Permissionless ElectionOpenness
         fn elect() -> Result<Supports<T::AccountId>, Self::Error> {
-            Self::kick_out_underperformed_non_reserved_validators();
+            Self::emit_fresh_bans_event();
+            let active_era = <T as Config>::EraInfoProvider::active_era().unwrap_or(0);
+            let ban_period = BanConfig::<T>::get().ban_period;
 
             let staking_validators = Self::DataProvider::electable_targets(None)
                 .map_err(Self::Error::DataProvider)?
                 .into_iter()
                 .collect::<BTreeSet<_>>();
-            let reserved_validators = NextEraReservedValidators::<T>::get()
+            let staking_reserved_validators = NextEraReservedValidators::<T>::get()
                 .into_iter()
+                .filter(|v| staking_validators.contains(v))
+                .collect::<BTreeSet<_>>();
+            let banned_validators = Banned::<T>::iter()
+                .filter(|(_, info)| info.start + ban_period >= active_era + 1)
```

Comment:
Here we can reuse `is_ban_lifted(ban_info, ban_period, active_era)` by **negating** its return value. ie `!is_ban_lifted(info, ban_period, active_era)`

Reply:
done